---
title: "CODE FOR FIGURE 3 AND SUPPLEMENTAL FIGURE 3"
author: "Matthew J Shannon"
date: "2024-12-17"
output: html_document
---

# Load in all dependencies

```{r, include = FALSE}
if (!requireNamespace ("BiocManager", quietly = TRUE))
    install.packages  ("BiocManager")
BiocManager::install  ("Rsamtools")
BiocManager::install  ("GenomicAlignments")
BiocManager::install  ("reshape2")
BiocManager::install  ("limma")
BiocManager::install  ("tidyverse")
BiocManager::install  ("reshape")
BiocManager::install  ("sva")
BiocManager::install  ("DESeq2")
BiocManager::install  ("biomaRt")
BiocManager::install  ("readxl")
BiocManager::install  ("apeglm")
BiocManager::install  ("clusterProfiler")
BiocManager::install  ("org.Hs.eg.db")
BiocManager::install  ("ggpubr")
BiocManager::install  ("ggpubr")
BiocManager::install  ("pvclust")
BiocManager::install  ('Seurat')
library               (Rsamtools)
library               (GenomicAlignments)
library               (GenomicRanges)
library               (tidyverse)
library               (knitr)
library               (readxl)
library               (edgeR)
library               (reshape2)
library               (limma)
library               (pheatmap)
library               (DESeq2)
library               (ggplot2)
library               (apeglm)
library               (grid)
library               (gridExtra)
library               (sva)
library               (png)
library               (biomaRt)
library               (RColorBrewer)
library               (ggpubr)
library               (pvclust)
library               (Seurat)
gc                    ()
```


# Load the necessary datasets

```{r}
# Full bulk RNA-seq (make sure sample E12 is removed) object
load         (file = "~/Desktop/FULLdata.Rdata")

# Early bulk RNA-seq (make sure sample E12 is removed) object
load         (file = "~/Desktop/EARLYdata.Rdata")

# Late bulk RNA-seq (make sure sample E12 is removed) object
load         (file = "~/Desktop/LATEdata.Rdata")

# Trophoblast single-cell object
load         (file = "~/Desktop/Trophoblast.Rdata")
DefaultAssay (Trophoblasts) <- "integrated"
```

# Figure 3

# Early first trimester (GA <10.0) sex-based differential expression analysis

```{r, Figure 3B, bulk RNA-seq early first trimester (GA <10.0) sex-based differential expression analysis prep}
EARLYdata$P185  <- as.integer (EARLYdata$P185)
EARLYdata$P156  <- as.integer (EARLYdata$P156)
EARLYdata$P223  <- as.integer (EARLYdata$P223)
EARLYdata$P766  <- as.integer (EARLYdata$P766)
EARLYdata$P280  <- as.integer (EARLYdata$P280)
EARLYdata$P5    <- as.integer (EARLYdata$P5)
EARLYdata$P15   <- as.integer (EARLYdata$P15)
EARLYdata$P250  <- as.integer (EARLYdata$P250)
EARLYdata$P568  <- as.integer (EARLYdata$P568)
EARLYdata$P193  <- as.integer (EARLYdata$P193)
EARLYdata$P217  <- as.integer (EARLYdata$P217)
EARLYdata$P181  <- as.integer (EARLYdata$P181)
EARLYdata$P247  <- as.integer (EARLYdata$P247)
EARLYdata$P252  <- as.integer (EARLYdata$P252)
EARLYdata$P456  <- as.integer (EARLYdata$P456)
EARLYdata$P979  <- as.integer (EARLYdata$P979)
EARLYdata$P984  <- as.integer (EARLYdata$P984)
EARLYdata$P996  <- as.integer (EARLYdata$P996)
EARLYdata$P1003 <- as.integer (EARLYdata$P1003)
EARLYdata$P1004 <- as.integer (EARLYdata$P1004)
EARLYdata$P1024 <- as.integer (EARLYdata$P1024)
EARLYdata$P1047 <- as.integer (EARLYdata$P1047)
EARLYdata$P1051 <- as.integer (EARLYdata$P1051)
counts          <- as.matrix  (EARLYdata)
meta            <- as.matrix  (metaData)
```

```{r, Figure 3B, bulk RNA-seq early first trimester (GA <10.0) sex-based differential expression analysis}
dds     <- DESeqDataSetFromMatrix (countData = counts, colData = meta, design = ~ Batch + Sex)
keep    <- rowSums                (counts (dds) >= 1) > 3
dds     <- dds [keep, ]

dds$Sex <- factor  (dds$Sex, levels = c("0", "1"))
dds     <- DESeq   (dds)
res     <- results (dds)
resSig  <- subset  (resOrdered, padj < 0.05)

view (resSig)
rm   (keep)
```

```{r, Figure 3B, bulk RNA-seq early first trimester (GA <10.0) volcano plot}
EnhancedVolcano (res,
                 lab             = rownames (res),
                 x               = "log2FoldChange",
                 y               = "pvalue",
                 FCcutoff        = 1.00,
                 pCutoff         = 10e-5,
                 pointSize       = 2.0,
                 col             = c('grey', 'grey', 'grey', 'red'),
                 colAlpha        = 1,
                 xlim            = c(-11, 11),
                 gridlines.major = FALSE,
                 drawConnectors  = FALSE, 
                 gridlines.minor = FALSE) + ggtitle (NULL)
```

```{r, Figure 3C, single-cell RNA-seq dotplot of the results from Fig 3B}
Trophoblasts$celltype.sex    <- paste       (Idents (Trophoblasts), Trophoblasts$Sex, sep = "_")
Trophoblasts$celltype        <- Idents      (Trophoblasts)

Idents (Trophoblasts) <- "celltype.sex"

DotPlot (Trophoblasts,
         features  = c("XIST",
                       "KDM6A",
                       "HSD17B10",
                       "KDM5C",
                       "HDHD1",
                       "RPS4X",
                       "SMC1A",
                       "DDX3X",
                       "UBA1",
                       "CD99",
                       "TTTY14",
                       "TTTY15",
                       "USP9Y",
                       "ZFY",
                       "TXLNGY",
                       "EIF1AY",
                       "KDM5D",
                       "DDX3Y",
                       "UTY",
                       "RPS4Y1"),
         assay     = "RNA",
         cols      = c("lightgrey", "red"),
         col.min   = 0,
         col.max   = 1,
         dot.min   = 0,
         dot.scale = 15) +
theme   (axis.text.x = element_text (angle = 90, hjust = 1))

Idents  (Trophoblasts) <- "celltype"
```

# Late first trimester (GA >=10.0) sex-based differential expression analysis

```{r, Figure 3D, bulk RNA-seq late first trimester (GA >=10.0) sex-based differential expression analysis prep}
LATEdata$P122  <- as.integer (LATEdata$P122)
LATEdata$P132  <- as.integer (LATEdata$P132)
LATEdata$P219  <- as.integer (LATEdata$P219)
LATEdata$P126  <- as.integer (LATEdata$P126)
LATEdata$P300  <- as.integer (LATEdata$P300)
LATEdata$P1012 <- as.integer (LATEdata$P1012)
LATEdata$P1015 <- as.integer (LATEdata$P1015)
LATEdata$P1019 <- as.integer (LATEdata$P1019)
LATEdata$P1025 <- as.integer (LATEdata$P1025)
LATEdata$P1029 <- as.integer (LATEdata$P1029)
LATEdata$P1037 <- as.integer (LATEdata$P1037)
LATEdata$P1041 <- as.integer (LATEdata$P1041)
counts          <- as.matrix  (LATEdata)
meta            <- as.matrix  (metaData)
```

```{r, Figure 3D, bulk RNA-seq late first trimester (GA >=10.0) sex-based differential expression analysis}
dds     <- DESeqDataSetFromMatrix (countData = counts, colData = meta, design = ~ Batch + Sex)
keep    <- rowSums                (counts (dds) >= 1) > 3
dds     <- dds [keep, ]

dds$Sex <- factor  (dds$Sex, levels = c("0", "1"))
dds     <- DESeq   (dds)
res     <- results (dds)
resSig  <- subset  (resOrdered, padj < 0.05)

view (resSig)
rm   (keep)
```

```{r, Figure 3D, bulk late first trimester (GA >=10.0) RNA-seq volcano plot}
EnhancedVolcano (res,
                 lab             = rownames (res),
                 x               = "log2FoldChange",
                 y               = "pvalue",
                 FCcutoff        = 1.00,
                 pCutoff         = 10e-5,
                 pointSize       = 2.0,
                 col             = c('grey', 'grey', 'grey', 'red'),
                 colAlpha        = 1,
                 xlim            = c(-11, 11),
                 gridlines.major = FALSE,
                 drawConnectors  = FALSE, 
                 gridlines.minor = FALSE) + ggtitle (NULL)
```

```{r, Figure 3E, single-cell RNA-seq dotplot of the results from Fig 3D}
Trophoblasts$celltype.sex    <- paste       (Idents (Trophoblasts), Trophoblasts$Sex, sep = "_")
Trophoblasts$celltype        <- Idents      (Trophoblasts)

Idents (Trophoblasts) <- "celltype.sex"

DotPlot (Trophoblasts,
         features  = c("C7orf71",
                       "APOL4",
                       "LOC101927354",
                       "ITGB3",
                       "LGALS13",
                       "PSG10P",
                       "C1QTNF1-AS1",
                       "ABCB1",
                       "PLEKHH1",
                       "LEP",
                       "C10orf10",
                       "LRP8",
                       "CEP85",
                       "MUC15",
                       "MAFF",
                       "HS3ST3B1",
                       "ANXA6",
                       "FAR2",
                       "PKIB",
                       "STK40",
                       "PTMA",
                       "PIGT",
                       "CD81",
                       "AMOTL2",
                       "SYT8",
                       "FADS3",
                       "ACVRL1",
                       "KLF6",
                       "FOS",
                       "EGR1",
                       "FOSB",
                       "CYR61"),
         assay     = "RNA",
         cols      = c("lightgrey", "red"),
         col.min   = 0,
         col.max   = 1,
         dot.min   = 0,
         dot.scale = 15) +
theme   (axis.text.x = element_text (angle = 90, hjust = 1))

Idents  (Trophoblasts) <- "celltype"
```

# Supplemental figure 3

```{r, Figure S3A, bulk RNA-seq (full dataset minus sample E12) sex-based differential expression analysis prep}
FULLdata$P185  <- as.integer (FULLdata$P185)
FULLdata$P156  <- as.integer (FULLdata$P156)
FULLdata$P223  <- as.integer (FULLdata$P223)
FULLdata$P766  <- as.integer (FULLdata$P766)
FULLdata$P280  <- as.integer (FULLdata$P280)
FULLdata$P5    <- as.integer (FULLdata$P5)
FULLdata$P15   <- as.integer (FULLdata$P15)
FULLdata$P250  <- as.integer (FULLdata$P250)
FULLdata$P568  <- as.integer (FULLdata$P568)
FULLdata$P193  <- as.integer (FULLdata$P193)
FULLdata$P217  <- as.integer (FULLdata$P217)
FULLdata$P181  <- as.integer (FULLdata$P181)
FULLdata$P247  <- as.integer (FULLdata$P247)
FULLdata$P252  <- as.integer (FULLdata$P252)
FULLdata$P456  <- as.integer (FULLdata$P456)
FULLdata$P979  <- as.integer (FULLdata$P979)
FULLdata$P984  <- as.integer (FULLdata$P984)
FULLdata$P996  <- as.integer (FULLdata$P996)
FULLdata$P1003 <- as.integer (FULLdata$P1003)
FULLdata$P1004 <- as.integer (FULLdata$P1004)
FULLdata$P1024 <- as.integer (FULLdata$P1024)
FULLdata$P1047 <- as.integer (FULLdata$P1047)
FULLdata$P1051 <- as.integer (FULLdata$P1051)
FULLdata$P122  <- as.integer (FULLdata$P122)
FULLdata$P132  <- as.integer (FULLdata$P132)
FULLdata$P219  <- as.integer (FULLdata$P219)
FULLdata$P126  <- as.integer (FULLdata$P126)
FULLdata$P300  <- as.integer (FULLdata$P300)
FULLdata$P1012 <- as.integer (FULLdata$P1012)
FULLdata$P1015 <- as.integer (FULLdata$P1015)
FULLdata$P1019 <- as.integer (FULLdata$P1019)
FULLdata$P1025 <- as.integer (FULLdata$P1025)
FULLdata$P1029 <- as.integer (FULLdata$P1029)
FULLdata$P1037 <- as.integer (FULLdata$P1037)
FULLdata$P1041 <- as.integer (FULLdata$P1041)
counts          <- as.matrix  (FULLdata)
meta            <- as.matrix  (metaData)
```

```{r, Figure S3A, bulk RNA-seq (full dataset minus sample E12) sex-based differential expression analysis}
dds     <- DESeqDataSetFromMatrix (countData = counts, colData = meta, design = ~ Batch + Sex)
keep    <- rowSums                (counts (dds) >= 1) > 3
dds     <- dds [keep, ]

dds$Sex <- factor  (dds$Sex, levels = c("0", "1"))
dds     <- DESeq   (dds)
res     <- results (dds)
resSig  <- subset  (resOrdered, padj < 0.05)

view (resSig)
rm   (keep)
```

```{r, S3A, bulk RNA-seq (full dataset minus sample E12) volcano plot}
EnhancedVolcano (res,
                 lab             = rownames (res),
                 x               = "log2FoldChange",
                 y               = "pvalue",
                 FCcutoff        = 1.00,
                 pCutoff         = 10e-5,
                 pointSize       = 2.0,
                 col             = c('grey', 'grey', 'grey', 'red'),
                 colAlpha        = 1,
                 xlim            = c(-11, 11),
                 gridlines.major = FALSE,
                 drawConnectors  = FALSE, 
                 gridlines.minor = FALSE) + ggtitle (NULL)
```

# Pseudotime analyses

```{r, S3B, single-cell RNA-seq pseudotime prep}
CTB_cCTB_EVT         <- subset              (Trophoblasts, idents = c("CTB", "cCTB", "EVT"))
CTB_cCTB_EVT.monocle <- as.CellDataSet      (CTB_cCTB_EVT, assay = "RNA")
CTB_cCTB_EVT.monocle <- estimateSizeFactors (CTB_cCTB_EVT.monocle)
CTB_cCTB_EVT.monocle <- estimateDispersions (CTB_cCTB_EVT.monocle)
CTB_cCTB_EVT.monocle <- detectGenes         (CTB_cCTB_EVT.monocle, min_expr = 0.1)

CTB_cCTB_EVT.monocle_expressed_genes <- row.names (subset (fData (CTB_cCTB_EVT.monocle), num_cells_expressed >= 10))

HLAG_id     <- row.names (subset (fData (CTB_cCTB_EVT.monocle), gene_short_name == "HLA-G"   ))
PAPPA2_id   <- row.names (subset (fData (CTB_cCTB_EVT.monocle), gene_short_name == "PAPPA2"  ))
MMP11_id    <- row.names (subset (fData (CTB_cCTB_EVT.monocle), gene_short_name == "MMP11"   ))
SERPINE2_id <- row.names (subset (fData (CTB_cCTB_EVT.monocle), gene_short_name == "SERPINE2"))

YAP1_id     <- row.names (subset (fData (CTB_cCTB_EVT.monocle), gene_short_name == "YAP1"    ))
SPINT1_id   <- row.names (subset (fData (CTB_cCTB_EVT.monocle), gene_short_name == "SPINT1"  ))
BCAM_id     <- row.names (subset (fData (CTB_cCTB_EVT.monocle), gene_short_name == "BCAM"    ))
ELF5_id     <- row.names (subset (fData (CTB_cCTB_EVT.monocle), gene_short_name == "ELF5"    ))

CTB_cCTB_EVT.monocle.2 <- newCellTypeHierarchy ()
CTB_cCTB_EVT.monocle.2 <- addCellType   (CTB_cCTB_EVT.monocle.2, "EVT",
                                         classify_func = function (x) { x [HLAG_id,     ]  >= 3.00 &
                                                                        x [PAPPA2_id,   ]  >= 1.00 &
                                                                        x [MMP11_id,    ]  >= 1.00 &
                                                                        x [SERPINE2_id, ]  >= 3.00   })
CTB_cCTB_EVT.monocle.2 <- addCellType   (CTB_cCTB_EVT.monocle.2, "CTB",
                                         classify_func = function (x) { x [YAP1_id,     ]  >= 1.00 &
                                                                        x [SPINT1_id,   ]  >= 2.00 &
                                                                        x [BCAM_id,     ]  >= 1.00 &
                                                                        x [ELF5_id,     ]  >= 0.50 &
                                                                        x [HLAG_id,     ]  <= 1.00 &
                                                                        x [PAPPA2_id,   ]  <= 0.50 &
                                                                        x [MMP11_id,    ]  <= 0.50 &
                                                                        x [SERPINE2_id, ]  <= 1.00   })
CTB_cCTB_EVT.monocle   <- classifyCells (CTB_cCTB_EVT.monocle, CTB_cCTB_EVT.monocle.2)

CTB_cCTB_EVT_marker_diff              <- markerDiffTable   (CTB_cCTB_EVT.monocle[CTB_cCTB_EVT.monocle_expressed_genes, ],
                                                            CTB_cCTB_EVT.monocle.2, cores = 4)
CTB_cCTB_EVT_semisup_clustering_genes <- row.names         (CTB_cCTB_EVT_marker_diff)[order (CTB_cCTB_EVT_marker_diff$qval)][1:250]
CTB_cCTB_EVT.monocle                  <- setOrderingFilter (CTB_cCTB_EVT.monocle, CTB_cCTB_EVT_semisup_clustering_genes)

rm (CTB_cCTB_EVT.monocle.2)
rm (CTB_cCTB_EVT_marker_diff)
rm (CTB_cCTB_EVT_semisup_clustering_genes)

CTB_cCTB_EVT.monocle <- reduceDimension (CTB_cCTB_EVT.monocle,
                                         max_components = 2,
                                         method         = 'DDRTree',
                                         norm_method    = 'none',
                                         pseudo_expr    = 0)
CTB_cCTB_EVT.monocle <- orderCells      (CTB_cCTB_EVT.monocle, root_state = 3)
```

```{r, S3B, single-cell RNA-seq pseudotime plots}
plot_cell_trajectory (CTB_cCTB_EVT.monocle,
                      cell_size       = 3,
                      color_by        = "seurat_clusters") +
theme                (legend.position = "right")

plot_cell_trajectory (CTB_cCTB_EVT.monocle,
                      cell_size       = 3,
                      color_by        = "Pseudotime") +
theme                (legend.position = "right")
```

```{r, S3C, single-cell RNA-seq gene expression across pseudotime prep}
CTB_cCTB_EVT_monocle <- AddMetaData (CTB_cCTB_EVT,
                                     metadata = CTB_cCTB_EVT.monocle@phenoData@data[["Pseudotime"]],
                                     col.name = "Pseudotime")

CTB_cCTB_EVT_XY_monocle <- subset (CTB_cCTB_EVT_monocle, subset = Sex == "Male")
CTB_cCTB_EVT_XX_monocle <- subset (CTB_cCTB_EVT_monocle, subset = Sex == "Female")
```

```{r, S3C, single-cell RNA-seq dotplot of CD99 gene expression across pseudotime}
Idents (CTB_cCTB_EVT_XY_monocle) <- 'celltype'
CTB_cCTB_EVT_XY_monocle2 <- RenameIdents (CTB_cCTB_EVT_XY_monocle, `CTB` = "darkorange1", `cCTB` = "red3", `EVT` = "forestgreen")

GENE_XY       <- as.data.frame      (CTB_cCTB_EVT_XY_monocle@assays$RNA@counts["CD99", ])
GENE_XY       <- rownames_to_column (GENE_XY,  "Cell")
State_XY      <- as.data.frame      (Idents (CTB_cCTB_EVT_XY_monocle2))
State_XY      <- rownames_to_column (State_XY, "Cell")
Plot_XY       <- merge              (GENE_XY, State_XY, by = "Cell")
Plot_XY       <- rownames_to_column (Plot_XY,  "#")
Pseudotime_XY <- as.data.frame      (CTB_cCTB_EVT_XY_monocle@meta.data$Pseudotime)
Pseudotime_XY <- rownames_to_column (Pseudotime_XY, "#")
Plot_XY       <- merge              (Plot_XY, Pseudotime_XY, by = "#")
Plot_XY       <- column_to_rownames (Plot_XY, "#")
Plot_XY       <- remove_rownames    (Plot_XY)
Plot_XY       <- column_to_rownames (Plot_XY, "Cell")
Plot_XY

Idents (CTB_cCTB_EVT_XX_monocle) <- 'celltype'
CTB_cCTB_EVT_XX_monocle2 <- RenameIdents (CTB_cCTB_EVT_XX_monocle, `CTB` = "darkorange1", `cCTB` = "red3", `EVT` = "forestgreen")

GENE_XX       <- as.data.frame      (CTB_cCTB_EVT_XX_monocle@assays$RNA@counts["CD99", ])
GENE_XX       <- rownames_to_column (GENE_XX,  "Cell")
State_XX      <- as.data.frame      (Idents (CTB_cCTB_EVT_XX_monocle2))
State_XX      <- rownames_to_column (State_XX, "Cell")
Plot_XX       <- merge              (GENE_XX, State_XX, by = "Cell")
Plot_XX       <- rownames_to_column (Plot_XX,  "#")
Pseudotime_XX <- as.data.frame      (CTB_cCTB_EVT_XX_monocle@meta.data$Pseudotime)
Pseudotime_XX <- rownames_to_column (Pseudotime_XX, "#")
Plot_XX       <- merge              (Plot_XX, Pseudotime_XX, by = "#")
Plot_XX       <- column_to_rownames (Plot_XX, "#")
Plot_XX       <- remove_rownames    (Plot_XX)
Plot_XX       <- column_to_rownames (Plot_XX, "Cell")
Plot_XX

ggplot      (Plot_XY,
             aes (x = CTB_cCTB_EVT_XY_monocle@meta.data$Pseudotime,
                  y = CTB_cCTB_EVT_XY_monocle@assays$RNA@counts["CD99", ])) +
geom_point  (colour = Idents(CTB_cCTB_EVT_XY_monocle2), size  = 2) +
geom_smooth (colour = "black", size = 1) +
xlim        (0, 22) +
xlab        ("Pseudotime") +
ylim        (0, 20) +
ylab        ("Relative Expression") +
theme       (axis.line  = element_line (size = 1, colour = "black"),
             axis.ticks = element_line (size = 1, colour = "black")) +
theme       (panel.background = element_blank ())

ggplot      (Plot_XX,
             aes (x = CTB_cCTB_EVT_XX_monocle@meta.data$Pseudotime,
                  y = CTB_cCTB_EVT_XX_monocle@assays$RNA@counts["CD99", ])) +
geom_point  (colour = Idents(CTB_cCTB_EVT_XX_monocle2), size  = 2) +
geom_smooth (colour = "black", size = 1) +
xlim        (0, 22) +
xlab        ("Pseudotime") +
ylim        (0, 20) +
ylab        ("Relative Expression") +
theme       (axis.line  = element_line (size = 1, colour = "black"),
             axis.ticks = element_line (size = 1, colour = "black")) +
theme       (panel.background = element_blank ())
```

```{r, S3C, single-cell RNA-seq dotplot of LEP gene expression across pseudotime}
Idents (CTB_cCTB_EVT_XY_monocle) <- 'celltype'
CTB_cCTB_EVT_XY_monocle2 <- RenameIdents (CTB_cCTB_EVT_XY_monocle, `CTB` = "darkorange1", `cCTB` = "red3", `EVT` = "forestgreen")

GENE_XY       <- as.data.frame      (CTB_cCTB_EVT_XY_monocle@assays$RNA@counts["LEP", ])
GENE_XY       <- rownames_to_column (GENE_XY,  "Cell")
State_XY      <- as.data.frame      (Idents (CTB_cCTB_EVT_XY_monocle2))
State_XY      <- rownames_to_column (State_XY, "Cell")
Plot_XY       <- merge              (GENE_XY, State_XY, by = "Cell")
Plot_XY       <- rownames_to_column (Plot_XY,  "#")
Pseudotime_XY <- as.data.frame      (CTB_cCTB_EVT_XY_monocle@meta.data$Pseudotime)
Pseudotime_XY <- rownames_to_column (Pseudotime_XY, "#")
Plot_XY       <- merge              (Plot_XY, Pseudotime_XY, by = "#")
Plot_XY       <- column_to_rownames (Plot_XY, "#")
Plot_XY       <- remove_rownames    (Plot_XY)
Plot_XY       <- column_to_rownames (Plot_XY, "Cell")
Plot_XY

Idents (CTB_cCTB_EVT_XX_monocle) <- 'celltype'
CTB_cCTB_EVT_XX_monocle2 <- RenameIdents (CTB_cCTB_EVT_XX_monocle, `CTB` = "darkorange1", `cCTB` = "red3", `EVT` = "forestgreen")

GENE_XX       <- as.data.frame      (CTB_cCTB_EVT_XX_monocle@assays$RNA@counts["LEP", ])
GENE_XX       <- rownames_to_column (GENE_XX,  "Cell")
State_XX      <- as.data.frame      (Idents (CTB_cCTB_EVT_XX_monocle2))
State_XX      <- rownames_to_column (State_XX, "Cell")
Plot_XX       <- merge              (GENE_XX, State_XX, by = "Cell")
Plot_XX       <- rownames_to_column (Plot_XX,  "#")
Pseudotime_XX <- as.data.frame      (CTB_cCTB_EVT_XX_monocle@meta.data$Pseudotime)
Pseudotime_XX <- rownames_to_column (Pseudotime_XX, "#")
Plot_XX       <- merge              (Plot_XX, Pseudotime_XX, by = "#")
Plot_XX       <- column_to_rownames (Plot_XX, "#")
Plot_XX       <- remove_rownames    (Plot_XX)
Plot_XX       <- column_to_rownames (Plot_XX, "Cell")
Plot_XX

ggplot      (Plot_XY,
             aes (x = CTB_cCTB_EVT_XY_monocle@meta.data$Pseudotime,
                  y = CTB_cCTB_EVT_XY_monocle@assays$RNA@counts["LEP", ])) +
geom_point  (colour = Idents(CTB_cCTB_EVT_XY_monocle2), size  = 2) +
geom_smooth (colour = "black", size = 1) +
xlim        (0, 22) +
xlab        ("Pseudotime") +
ylim        (0, 20) +
ylab        ("Relative Expression") +
theme       (axis.line  = element_line (size = 1, colour = "black"),
             axis.ticks = element_line (size = 1, colour = "black")) +
theme       (panel.background = element_blank ())

ggplot      (Plot_XX,
             aes (x = CTB_cCTB_EVT_XX_monocle@meta.data$Pseudotime,
                  y = CTB_cCTB_EVT_XX_monocle@assays$RNA@counts["LEP", ])) +
geom_point  (colour = Idents(CTB_cCTB_EVT_XX_monocle2), size  = 2) +
geom_smooth (colour = "black", size = 1) +
xlim        (0, 22) +
xlab        ("Pseudotime") +
ylim        (0, 20) +
ylab        ("Relative Expression") +
theme       (axis.line  = element_line (size = 1, colour = "black"),
             axis.ticks = element_line (size = 1, colour = "black")) +
theme       (panel.background = element_blank ())
```

```{r, S3C, single-cell RNA-seq dotplot of AMOTL2 gene expression across pseudotime}
Idents (CTB_cCTB_EVT_XY_monocle) <- 'celltype'
CTB_cCTB_EVT_XY_monocle2 <- RenameIdents (CTB_cCTB_EVT_XY_monocle, `CTB` = "darkorange1", `cCTB` = "red3", `EVT` = "forestgreen")

GENE_XY       <- as.data.frame      (CTB_cCTB_EVT_XY_monocle@assays$RNA@counts["AMOTL2", ])
GENE_XY       <- rownames_to_column (GENE_XY,  "Cell")
State_XY      <- as.data.frame      (Idents (CTB_cCTB_EVT_XY_monocle2))
State_XY      <- rownames_to_column (State_XY, "Cell")
Plot_XY       <- merge              (GENE_XY, State_XY, by = "Cell")
Plot_XY       <- rownames_to_column (Plot_XY,  "#")
Pseudotime_XY <- as.data.frame      (CTB_cCTB_EVT_XY_monocle@meta.data$Pseudotime)
Pseudotime_XY <- rownames_to_column (Pseudotime_XY, "#")
Plot_XY       <- merge              (Plot_XY, Pseudotime_XY, by = "#")
Plot_XY       <- column_to_rownames (Plot_XY, "#")
Plot_XY       <- remove_rownames    (Plot_XY)
Plot_XY       <- column_to_rownames (Plot_XY, "Cell")
Plot_XY

Idents (CTB_cCTB_EVT_XX_monocle) <- 'celltype'
CTB_cCTB_EVT_XX_monocle2 <- RenameIdents (CTB_cCTB_EVT_XX_monocle, `CTB` = "darkorange1", `cCTB` = "red3", `EVT` = "forestgreen")

GENE_XX       <- as.data.frame      (CTB_cCTB_EVT_XX_monocle@assays$RNA@counts["AMOTL2", ])
GENE_XX       <- rownames_to_column (GENE_XX,  "Cell")
State_XX      <- as.data.frame      (Idents (CTB_cCTB_EVT_XX_monocle2))
State_XX      <- rownames_to_column (State_XX, "Cell")
Plot_XX       <- merge              (GENE_XX, State_XX, by = "Cell")
Plot_XX       <- rownames_to_column (Plot_XX,  "#")
Pseudotime_XX <- as.data.frame      (CTB_cCTB_EVT_XX_monocle@meta.data$Pseudotime)
Pseudotime_XX <- rownames_to_column (Pseudotime_XX, "#")
Plot_XX       <- merge              (Plot_XX, Pseudotime_XX, by = "#")
Plot_XX       <- column_to_rownames (Plot_XX, "#")
Plot_XX       <- remove_rownames    (Plot_XX)
Plot_XX       <- column_to_rownames (Plot_XX, "Cell")
Plot_XX

ggplot      (Plot_XY,
             aes (x = CTB_cCTB_EVT_XY_monocle@meta.data$Pseudotime,
                  y = CTB_cCTB_EVT_XY_monocle@assays$RNA@counts["AMOTL2", ])) +
geom_point  (colour = Idents(CTB_cCTB_EVT_XY_monocle2), size  = 2) +
geom_smooth (colour = "black", size = 1) +
xlim        (0, 22) +
xlab        ("Pseudotime") +
ylim        (0, 20) +
ylab        ("Relative Expression") +
theme       (axis.line  = element_line (size = 1, colour = "black"),
             axis.ticks = element_line (size = 1, colour = "black")) +
theme       (panel.background = element_blank ())

ggplot      (Plot_XX,
             aes (x = CTB_cCTB_EVT_XX_monocle@meta.data$Pseudotime,
                  y = CTB_cCTB_EVT_XX_monocle@assays$RNA@counts["AMOTL2", ])) +
geom_point  (colour = Idents(CTB_cCTB_EVT_XX_monocle2), size  = 2) +
geom_smooth (colour = "black", size = 1) +
xlim        (0, 22) +
xlab        ("Pseudotime") +
ylim        (0, 20) +
ylab        ("Relative Expression") +
theme       (axis.line  = element_line (size = 1, colour = "black"),
             axis.ticks = element_line (size = 1, colour = "black")) +
theme       (panel.background = element_blank ())
```

# Gene expression across week of GA

```{r, S3D, bulk RNA-seq (full dataset minus sample E12) bar plots}
# Ensure metadata has a GA2 column with gestational age binned into week of gestation categories and the dds object is created from the FULLdata object without sample E12

      C7orf71 <- plotCounts (dds, gene = "C7orf71", intgroup = c("Sex", "GA2"), returnData = TRUE)
      C7orf71 <- arrange    (C7orf71, Sex, GA2)
view (C7orf71)   # exported and plotted using GraphPad Prism

      APOL4 <- plotCounts (dds, gene = "APOL4", intgroup = c("Sex", "GA2"), returnData = TRUE)
      APOL4 <- arrange    (APOL4, Sex, GA2)
view (APOL4)   # exported and plotted using GraphPad Prism

      LOC101927354 <- plotCounts (dds, gene = "LOC101927354", intgroup = c("Sex", "GA2"), returnData = TRUE)
      LOC101927354 <- arrange    (LOC101927354, Sex, GA2)
view (LOC101927354)   # exported and plotted using GraphPad Prism

      ITGB3 <- plotCounts (dds, gene = "ITGB3", intgroup = c("Sex", "GA2"), returnData = TRUE)
      ITGB3 <- arrange    (ITGB3, Sex, GA2)
view (ITGB3)   # exported and plotted using GraphPad Prism

      LGALS13 <- plotCounts (dds, gene = "LGALS13", intgroup = c("Sex", "GA2"), returnData = TRUE)
      LGALS13 <- arrange    (LGALS13, Sex, GA2)
view (LGALS13)   # exported and plotted using GraphPad Prism

      PSG10P <- plotCounts (dds, gene = "PSG10P", intgroup = c("Sex", "GA2"), returnData = TRUE)
      PSG10P <- arrange    (PSG10P, Sex, GA2)
view (PSG10P)   # exported and plotted using GraphPad Prism

      C1QTNF1-AS1 <- plotCounts (dds, gene = "C1QTNF1-AS1", intgroup = c("Sex", "GA2"), returnData = TRUE)
      C1QTNF1-AS1 <- arrange    (C1QTNF1-AS1, Sex, GA2)
view (C1QTNF1-AS1)   # exported and plotted using GraphPad Prism

      ABCB1 <- plotCounts (dds, gene = "ABCB1", intgroup = c("Sex", "GA2"), returnData = TRUE)
      ABCB1 <- arrange    (ABCB1, Sex, GA2)
view (ABCB1)   # exported and plotted using GraphPad Prism

      PLEKHH1 <- plotCounts (dds, gene = "PLEKHH1", intgroup = c("Sex", "GA2"), returnData = TRUE)
      PLEKHH1 <- arrange    (PLEKHH1, Sex, GA2)
view (PLEKHH1)   # exported and plotted using GraphPad Prism

      LEP <- plotCounts (dds, gene = "LEP", intgroup = c("Sex", "GA2"), returnData = TRUE)
      LEP <- arrange    (LEP, Sex, GA2)
view (LEP)   # exported and plotted using GraphPad Prism

      C10orf10 <- plotCounts (dds, gene = "C10orf10", intgroup = c("Sex", "GA2"), returnData = TRUE)
      C10orf10 <- arrange    (C10orf10, Sex, GA2)
view (C10orf10)   # exported and plotted using GraphPad Prism

      LRP8 <- plotCounts (dds, gene = "LRP8", intgroup = c("Sex", "GA2"), returnData = TRUE)
      LRP8 <- arrange    (LRP8, Sex, GA2)
view (LRP8)   # exported and plotted using GraphPad Prism

      CEP85 <- plotCounts (dds, gene = "CEP85", intgroup = c("Sex", "GA2"), returnData = TRUE)
      CEP85 <- arrange    (CEP85, Sex, GA2)
view (CEP85)   # exported and plotted using GraphPad Prism

      MUC15 <- plotCounts (dds, gene = "MUC15", intgroup = c("Sex", "GA2"), returnData = TRUE)
      MUC15 <- arrange    (MUC15, Sex, GA2)
view (MUC15)   # exported and plotted using GraphPad Prism

      MAFF <- plotCounts (dds, gene = "MAFF", intgroup = c("Sex", "GA2"), returnData = TRUE)
      MAFF <- arrange    (MAFF, Sex, GA2)
view (MAFF)   # exported and plotted using GraphPad Prism

      HS3ST3B1 <- plotCounts (dds, gene = "HS3ST3B1", intgroup = c("Sex", "GA2"), returnData = TRUE)
      HS3ST3B1 <- arrange    (HS3ST3B1, Sex, GA2)
view (HS3ST3B1)   # exported and plotted using GraphPad Prism

      ANXA6 <- plotCounts (dds, gene = "ANXA6", intgroup = c("Sex", "GA2"), returnData = TRUE)
      ANXA6 <- arrange    (ANXA6, Sex, GA2)
view (ANXA6)   # exported and plotted using GraphPad Prism

      FAR2 <- plotCounts (dds, gene = "FAR2", intgroup = c("Sex", "GA2"), returnData = TRUE)
      FAR2 <- arrange    (FAR2, Sex, GA2)
view (FAR2)   # exported and plotted using GraphPad Prism

      PKIB <- plotCounts (dds, gene = "PKIB", intgroup = c("Sex", "GA2"), returnData = TRUE)
      PKIB <- arrange    (PKIB, Sex, GA2)
view (PKIB)   # exported and plotted using GraphPad Prism

      STK40 <- plotCounts (dds, gene = "STK40", intgroup = c("Sex", "GA2"), returnData = TRUE)
      STK40 <- arrange    (STK40, Sex, GA2)
view (STK40)   # exported and plotted using GraphPad Prism

      PTMA <- plotCounts (dds, gene = "PTMA", intgroup = c("Sex", "GA2"), returnData = TRUE)
      PTMA <- arrange    (PTMA, Sex, GA2)
view (PTMA)   # exported and plotted using GraphPad Prism

      PIGT <- plotCounts (dds, gene = "PIGT", intgroup = c("Sex", "GA2"), returnData = TRUE)
      PIGT <- arrange    (PIGT, Sex, GA2)
view (PIGT)   # exported and plotted using GraphPad Prism

      CD81 <- plotCounts (dds, gene = "CD81", intgroup = c("Sex", "GA2"), returnData = TRUE)
      CD81 <- arrange    (CD81, Sex, GA2)
view (CD81)   # exported and plotted using GraphPad Prism

      AMOTL2 <- plotCounts (dds, gene = "AMOTL2", intgroup = c("Sex", "GA2"), returnData = TRUE)
      AMOTL2 <- arrange    (AMOTL2, Sex, GA2)
view (AMOTL2)   # exported and plotted using GraphPad Prism

      SYT8 <- plotCounts (dds, gene = "SYT8", intgroup = c("Sex", "GA2"), returnData = TRUE)
      SYT8 <- arrange    (SYT8, Sex, GA2)
view (SYT8)   # exported and plotted using GraphPad Prism

      FADS3 <- plotCounts (dds, gene = "FADS3", intgroup = c("Sex", "GA2"), returnData = TRUE)
      FADS3 <- arrange    (FADS3, Sex, GA2)
view (FADS3)   # exported and plotted using GraphPad Prism

      ACVRL1 <- plotCounts (dds, gene = "ACVRL1", intgroup = c("Sex", "GA2"), returnData = TRUE)
      ACVRL1 <- arrange    (ACVRL1, Sex, GA2)
view (ACVRL1)   # exported and plotted using GraphPad Prism

      KLF6 <- plotCounts (dds, gene = "KLF6", intgroup = c("Sex", "GA2"), returnData = TRUE)
      KLF6 <- arrange    (KLF6, Sex, GA2)
view (KLF6)   # exported and plotted using GraphPad Prism

      FOS <- plotCounts (dds, gene = "FOS", intgroup = c("Sex", "GA2"), returnData = TRUE)
      FOS <- arrange    (FOS, Sex, GA2)
view (FOS)   # exported and plotted using GraphPad Prism

      EGR1 <- plotCounts (dds, gene = "EGR1", intgroup = c("Sex", "GA2"), returnData = TRUE)
      EGR1 <- arrange    (EGR1, Sex, GA2)
view (EGR1)   # exported and plotted using GraphPad Prism

      FOSB <- plotCounts (dds, gene = "FOSB", intgroup = c("Sex", "GA2"), returnData = TRUE)
      FOSB <- arrange    (FOSB, Sex, GA2)
view (FOSB)   # exported and plotted using GraphPad Prism

      CYR61 <- plotCounts (dds, gene = "CYR61", intgroup = c("Sex", "GA2"), returnData = TRUE)
      CYR61 <- arrange    (CYR61, Sex, GA2)
view (CYR61)   # exported and plotted using GraphPad Prism
```
