---
title: "CODE FOR FIGURE 4 AND SUPPLEMENTAL FIGURE 4"
author: "Matthew J Shannon"
date: "2024-12-17"
output: html_document
---

# Load in all dependencies

```{r, include = FALSE}
devtools::install_github  ('chris-mcginnis-ucsf/DoubletFinder')
devtools::install_github  ('satijalab/seurat-wrappers')
remotes::install_github   ("mojaveazure/seurat-disk")
devtools::install_github  ("velocyto-team/velocyto.R")
devtools::install_github  ("jinworks/CellChat")
BiocManager::install      ("pcaMethods")
BiocManager::install      ('MAST')
BiocManager::install      ('org.Hs.eg.db')
BiocManager::install      (version = "3.12")
BiocManager::install      ('Seurat')
BiocManager::install      ('readxl')
BiocManager::install      ('modes')
BiocManager::install      ('pheatmap')
BiocManager::install      ('limma')
BiocManager::install      ('clustree')
BiocManager::install      ('clusterProfiler')
BiocManager::install      ('EnhancedVolcano')
install.packages          ('fields')
install.packages          ('plotly')
install.packages          ("VennDiagram")
library                   (Seurat)
library                   (readr)
library                   (tidyverse)
library                   (Matrix)
library                   (dplyr)
library                   (ggplot2)
library                   (cowplot)
library                   (tibble)
library                   (readxl)
library                   (sctransform)
library                   (fields)
library                   (KernSmooth)
library                   (ROCR)
library                   (parallel)
library                   (reshape2)
library                   (pheatmap)
library                   (DoubletFinder)
library                   (limma)
library                   (SeuratWrappers)
library                   (SeuratDisk)
library                   (plotly)
library                   (clustree)
library                   (velocyto.R)
library                   (MAST)
library                   (EnhancedVolcano)
library                   (clusterProfiler)
library                   (AnnotationDbi)
library                   (org.Hs.eg.db)
library                   (VennDiagram)
library                   (CellChat)
gc                        ()
```

# Load the necessary datasets

```{r}
# Trophoblast single-cell object
load         (file = "~/Desktop/Trophoblast.Rdata")
DefaultAssay (Trophoblasts) <- "integrated"

# Expanded trophoblast single-cell object
load         (file = "~/Desktop/Expanded_Trophoblast_Data.Rdata")
DefaultAssay (Expanded_Trophoblasts) <- "integrated"
```

# Figure 4

```{r, Figure 4A, expanded single-cell RNA-seq UMAP}
Idents (Expanded_Trophoblasts) <- "celltype"

UMAPPlot (Expanded_Trophoblasts,
          split.by = "Sex",
          label    = TRUE,
          pt.size  = 2,
          cols     = c('CTB'               = 'darkorange1',
                       'cCTB'              = 'red3',
                       'EVT'               = 'forestgreen',
                       'SCTp'              = 'dodgerblue4',
                       'Macrophage'        = 'firebrick1',
                       'uNK'               = 'purple1',
                       'cNK'               = 'purple4',
                       'Epiglandular cell' = 'coral1',
                       'Endothelial cell'  = 'turquoise2',
                       'Stromal cell'      = 'steelblue3')) + NoLegend ()
```

```{r, Cell Chat Analysis - EVT-Uterine Network from the expanded single-cell RNA-seq data}
Idents (Expanded_Trophoblasts) <- "celltype"

Interaction        <- subset (Expanded_Trophoblasts, idents = c("EVT", "Endothelial cell", "Stromal cell", "Epiglandular cell", "Macrophage", "uNK", "cNK"))

CellChatDB         <- CellChatDB.human
CellChatDB.use     <- CellChatDB

cellChat_MALE@DB   <- CellChatDB.use
cellChat_FEMALE@DB <- CellChatDB.use

Male_Interaction              <- subset                            (Interaction, subset = Sex == "Male")
MALE.small                    <- subset                            (Male_Interaction, downsample = 500)
MALE.small@meta.data$celltype <- droplevels                        (MALE.small@meta.data$celltype)
cellChat_MALE                 <- createCellChat                    (MALE.small, group.by = "celltype", assay = "RNA")
cellChat_MALE                 <- subsetData                        (cellChat_MALE)                                         # Necessary, even when using the whole database
cellChat_MALE                 <- identifyOverExpressedGenes        (cellChat_MALE)
cellChat_MALE                 <- identifyOverExpressedInteractions (cellChat_MALE)
cellChat_MALE                 <- projectData                       (cellChat_MALE, PPI.human)                              # Optional
cellChat_MALE                 <- computeCommunProb                 (cellChat_MALE, type = "triMean")
cellChat_MALE                 <- filterCommunication               (cellChat_MALE, min.cells = 3)
MALE_df.net1                  <- subsetCommunication               (cellChat_MALE)
MALE_df.net2                  <- subsetCommunication               (cellChat_MALE, slot.name = "netP")
cellChat_MALE                 <- computeCommunProbPathway          (cellChat_MALE)
cellChat_MALE                 <- aggregateNet                      (cellChat_MALE)

Female_Interaction              <- subset                            (Interaction, subset = Sex == "Female")
FEMALE.small                    <- subset                            (Female_Interaction, downsample = 500)
FEMALE.small@meta.data$celltype <- droplevels                        (FEMALE.small@meta.data$celltype)
cellChat_FEMALE                 <- createCellChat                    (FEMALE.small, group.by = "celltype", assay = "RNA")
cellChat_FEMALE                 <- subsetData                        (cellChat_FEMALE)                                     # Necessary, even when using the whole database
cellChat_FEMALE                 <- identifyOverExpressedGenes        (cellChat_FEMALE)
cellChat_FEMALE                 <- identifyOverExpressedInteractions (cellChat_FEMALE)
cellChat_FEMALE                 <- projectData                       (cellChat_FEMALE, PPI.human)                          # Optional
cellChat_FEMALE                 <- computeCommunProb                 (cellChat_FEMALE, type = "triMean")
cellChat_FEMALE                 <- filterCommunication               (cellChat_FEMALE, min.cells = 3)
FEMALE_df.net1                  <- subsetCommunication               (cellChat_FEMALE)
FEMALE_df.net2                  <- subsetCommunication               (cellChat_FEMALE, slot.name = "netP")
cellChat_FEMALE                 <- computeCommunProbPathway          (cellChat_FEMALE)
cellChat_FEMALE                 <- aggregateNet                      (cellChat_FEMALE)

object.list <- list          (MALE = cellChat_MALE, FEMALE = cellChat_FEMALE)
cellchat    <- mergeCellChat (object.list, add.names = names (object.list), cell.prefix = TRUE)

gc ()

# Table S5
write.table (MALE_df.net1, sep = "\t",   file = "Male_Network_1.txt")
write.table (MALE_df.net2, sep = "\t",   file = "Male_Network_2.txt")

write.table (FEMALE_df.net1, sep = "\t", file = "Female_Network_1.txt")
write.table (FEMALE_df.net2, sep = "\t", file = "Female_Network_2.txt")

# Figure 4B (EVT-Uterine Network)
compareInteractions (cellchat, show.legend = F, group = c(1,2)) + ylim (0, 1000)
```

```{r, Figure 4C, EVT-uterine network MIF circle plot}
# Circle plot
pathways.show <- c("MIF") 
weight.max    <- getMaxWeight (object.list, slot.name = c("netP"), attribute = pathways.show) # control the edge weights across different data-sets

# Figure 4C (EVT-Uterine Network)
for     (i in 1: length (object.list)) {
netVisual_aggregate (object.list[[i]],
                     signaling       = pathways.show,
                     layout          = "circle",
                     edge.weight.max = weight.max[1],
                     edge.width.max  = 10,
                     color.use = c('uNK'               = 'purple1',
                                   'EVT'               = 'forestgreen',
                                   'Stromal cell'      = 'steelblue3',
                                   'Macrophage'        = 'firebrick1',
                                   'cNK'               = 'purple4',
                                   'Endothelial cell'  = 'turquoise2',
                                   'Epiglandular cell' = 'coral1'),
                     signaling.name  = paste (pathways.show, names (object.list)[i])) }

# Figure 4E (EVT-Uterine Network)
netAnalysis_contribution (object.list[[i]], signaling = pathways.show)
```

```{r, Figure 4C, EVT-uterine network CD99 circle plot}
# Circle plot
pathways.show <- c("CD99") 
weight.max    <- getMaxWeight (object.list, slot.name = c("netP"), attribute = pathways.show) # control the edge weights across different data-sets

# Figure 4C (EVT-Uterine Network)
for     (i in 1: length (object.list)) {
netVisual_aggregate (object.list[[i]],
                     signaling       = pathways.show,
                     layout          = "circle",
                     edge.weight.max = weight.max[1],
                     edge.width.max  = 10,
                     color.use = c('uNK'               = 'purple1',
                                   'EVT'               = 'forestgreen',
                                   'Stromal cell'      = 'steelblue3',
                                   'Macrophage'        = 'firebrick1',
                                   'cNK'               = 'purple4',
                                   'Endothelial cell'  = 'turquoise2',
                                   'Epiglandular cell' = 'coral1'),
                     signaling.name  = paste (pathways.show, names (object.list)[i])) }

# Figure 4E (EVT-Uterine Network)
netAnalysis_contribution (object.list[[i]], signaling = pathways.show)
```

```{r, Figure 4C, EVT-uterine network VEGF circle plot}
# Circle plot
pathways.show <- c("VEGF") 
weight.max    <- getMaxWeight (object.list, slot.name = c("netP"), attribute = pathways.show) # control the edge weights across different data-sets

# Figure 4C (EVT-Uterine Network)
for     (i in 1: length (object.list)) {
netVisual_aggregate (object.list[[i]],
                     signaling       = pathways.show,
                     layout          = "circle",
                     edge.weight.max = weight.max[1],
                     edge.width.max  = 10,
                     color.use = c('uNK'               = 'purple1',
                                   'EVT'               = 'forestgreen',
                                   'Stromal cell'      = 'steelblue3',
                                   'Macrophage'        = 'firebrick1',
                                   'cNK'               = 'purple4',
                                   'Endothelial cell'  = 'turquoise2',
                                   'Epiglandular cell' = 'coral1'),
                     signaling.name  = paste (pathways.show, names (object.list)[i])) }

# Figure 4E (EVT-Uterine Network)
netAnalysis_contribution (object.list[[i]], signaling = pathways.show)
```

```{r, Cell Chat Analysis - Trophoblast Network from the expanded single-cell RNA-seq data}
Idents (Expanded_Trophoblasts) <- "celltype"

Interaction        <- subset (Expanded_Trophoblasts, idents = c("CTB", "cCTB", "EVT", "SCTp"))

CellChatDB         <- CellChatDB.human
CellChatDB.use     <- CellChatDB

cellChat_MALE@DB   <- CellChatDB.use
cellChat_FEMALE@DB <- CellChatDB.use

Male_Interaction              <- subset                            (Interaction, subset = Sex == "Male")
MALE.small                    <- subset                            (Male_Interaction,   downsample = 500)
MALE.small@meta.data$celltype <- droplevels                        (MALE.small@meta.data$celltype)
cellChat_MALE                 <- createCellChat                    (MALE.small,   group.by = "celltype", assay = "RNA")
cellChat_MALE                 <- subsetData                        (cellChat_MALE)                                         # Necessary, even when using the whole database
cellChat_MALE                 <- identifyOverExpressedGenes        (cellChat_MALE)
cellChat_MALE                 <- identifyOverExpressedInteractions (cellChat_MALE)
cellChat_MALE                 <- projectData                       (cellChat_MALE,   PPI.human)                            # Optional
cellChat_MALE                 <- computeCommunProb                 (cellChat_MALE,   type = "triMean")
cellChat_MALE                 <- filterCommunication               (cellChat_MALE,   min.cells = 3)
MALE_df.net1                  <- subsetCommunication               (cellChat_MALE)
MALE_df.net2                  <- subsetCommunication               (cellChat_MALE,   slot.name = "netP")
cellChat_MALE                 <- computeCommunProbPathway          (cellChat_MALE)
cellChat_MALE                 <- aggregateNet                      (cellChat_MALE)

Female_Interaction              <- subset                            (Interaction, subset = Sex == "Female")
FEMALE.small                    <- subset                            (Female_Interaction, downsample = 500)
FEMALE.small@meta.data$celltype <- droplevels                        (FEMALE.small@meta.data$celltype)
cellChat_FEMALE                 <- createCellChat                    (FEMALE.small, group.by = "celltype", assay = "RNA")
cellChat_FEMALE                 <- subsetData                        (cellChat_FEMALE)                                     # Necessary, even when using the whole database
cellChat_FEMALE                 <- identifyOverExpressedGenes        (cellChat_FEMALE)
cellChat_FEMALE                 <- identifyOverExpressedInteractions (cellChat_FEMALE)
cellChat_FEMALE                 <- projectData                       (cellChat_FEMALE, PPI.human)                          # Optional
cellChat_FEMALE                 <- computeCommunProb                 (cellChat_FEMALE, type = "triMean")
cellChat_FEMALE                 <- filterCommunication               (cellChat_FEMALE, min.cells = 3)
FEMALE_df.net1                  <- subsetCommunication               (cellChat_FEMALE)
FEMALE_df.net2                  <- subsetCommunication               (cellChat_FEMALE, slot.name = "netP")
cellChat_FEMALE                 <- computeCommunProbPathway          (cellChat_FEMALE)
cellChat_FEMALE                 <- aggregateNet                      (cellChat_FEMALE)

object.list <- list          (MALE = cellChat_MALE, FEMALE = cellChat_FEMALE)
cellchat    <- mergeCellChat (object.list, add.names = names (object.list), cell.prefix = TRUE)

gc ()

# Table S5
write.table (MALE_df.net1, sep = "\t",   file = "TB_Male_Network_1.txt")
write.table (MALE_df.net2, sep = "\t",   file = "TB_Male_Network_2.txt")

write.table (FEMALE_df.net1, sep = "\t", file = "TB_Female_Network_1.txt")
write.table (FEMALE_df.net2, sep = "\t", file = "TB_Female_Network_2.txt")

# Figure 4B (Trophoblast Network)
compareInteractions (cellchat, show.legend = F, group = c(1,2)) + ylim (0, 1000)
```

```{r, Figure 4C, trophoblast network CD99 circle plot}
# Circle plot
pathways.show <- c("CD99") 
weight.max    <- getMaxWeight (object.list, slot.name = c("netP"), attribute = pathways.show) # control the edge weights across different data-sets


# Figure 4C (Trophoblast Network)
for     (i in 1: length (object.list)) {
netVisual_aggregate (object.list[[i]],
                     signaling       = pathways.show,
                     layout          = "circle",
                     edge.weight.max = weight.max[1],
                     edge.width.max  = 10,
                     color.use = c('CTB'  = 'darkorange1',
                                   'EVT'  = 'forestgreen',
                                   'cCTB' = 'red3',
                                   'SCTp' = 'dodgerblue4'),
                     signaling.name  = paste (pathways.show, names (object.list)[i])) }

# Figure 4E (Trophoblast Network)
netAnalysis_contribution (object.list[[i]], signaling = pathways.show)
```

```{r, Figure 4C, trophoblast network VEGF circle plot}
# Circle plot
pathways.show <- c("VEGF") 
weight.max    <- getMaxWeight (object.list, slot.name = c("netP"), attribute = pathways.show) # control the edge weights across different data-sets


# Figure 4C (Trophoblast Network)
for     (i in 1: length (object.list)) {
netVisual_aggregate (object.list[[i]],
                     signaling       = pathways.show,
                     layout          = "circle",
                     edge.weight.max = weight.max[1],
                     edge.width.max  = 10,
                     color.use = c('CTB'  = 'darkorange1',
                                   'EVT'  = 'forestgreen',
                                   'cCTB' = 'red3',
                                   'SCTp' = 'dodgerblue4'),
                     signaling.name  = paste (pathways.show, names (object.list)[i])) }

# Figure 4E (Trophoblast Network)
netAnalysis_contribution (object.list[[i]], signaling = pathways.show)
```

```{r, Figure 4D, expanded single-cell RNA-seq dotplot of MIF and VEGF signalling components}
Expanded_Trophoblasts$celltype.sex <- paste  (Idents (Expanded_Trophoblasts), Expanded_Trophoblasts$Sex, sep = "_")
Expanded_Trophoblasts$celltype     <- Idents (Expanded_Trophoblasts)

Idents (Expanded_Trophoblasts) <- "celltype.sex"

my_levels <- c("CTB_Male",               "CTB_Female",
               "cCTB_Male",              "cCTB_Female",
               "EVT_Male",               "EVT_Female",
               "SCTp_Male",              "SCTp_Female",
               "Macrophage_Male",        "Macrophage_Female",
               "uNK_Male",               "uNK_Female",
               "cNK_Male",               "cNK_Female",
               "Epiglandular cell_Male", "Epiglandular cell_Female",
               "Stromal cell_Male",      "Stromal cell_Female",
               "Endothelial cell_Male",  "Endothelial cell_Female")

Idents (Expanded_Trophoblasts) <- factor (Expanded_Trophoblasts$celltype.sex, levels = my_levels)

DotPlot (Expanded_Trophoblasts,
         features  = c("MIF",
                       "CD74",
                       "CD44",
                       "CXCR2",
                       "CXCR4",
                       "VEGFA",
                       "PGF",
                       "FLT1",
                       "KDR",
                       "FLT4"),
         assay     = "RNA",
         cols      = c("lightgrey", "red"),
         col.min   = 0,
         col.max   = 1,
         dot.min   = 0,
         dot.scale = 15) +
theme   (axis.text.x = element_text (angle = 90, hjust = 1))

Idents (Expanded_Trophoblasts) <- "celltype"
```

# Supplemental figure 4

```{r, Figure S4A, expanded single-cell RNA-seq dotplot of MPI-associated gene markers}
Expanded_Trophoblasts$celltype.sex <- paste  (Idents (Expanded_Trophoblasts), Expanded_Trophoblasts$Sex, sep = "_")
Expanded_Trophoblasts$celltype     <- Idents (Expanded_Trophoblasts)

Idents (Expanded_Trophoblasts) <- "celltype.sex"

my_levels <- c("CTB_Male",               "CTB_Female",
               "cCTB_Male",              "cCTB_Female",
               "EVT_Male",               "EVT_Female",
               "SCTp_Male",              "SCTp_Female",
               "Macrophage_Male",        "Macrophage_Female",
               "uNK_Male",               "uNK_Female",
               "cNK_Male",               "cNK_Female",
               "Epiglandular cell_Male", "Epiglandular cell_Female",
               "Stromal cell_Male",      "Stromal cell_Female",
               "Endothelial cell_Male",  "Endothelial cell_Female")

Idents (Expanded_Trophoblasts) <- factor (Expanded_Trophoblasts$celltype.sex, levels = my_levels)

DotPlot (Expanded_Trophoblasts,
         features  = c("CD34",
                       "KIT",
                       "ENTPD1",
                       "NCAM1",
                       "ITGAX",
                       "PTPRC",
                       "KLRB1",
                       "CD8A",
                       "IL7R",
                       "CD3G",
                       "FOXP3",
                       "FCER1A",
                       "CD1C",
                       "CLEC9A",
                       "S100A12",
                       "IL1B",
                       "FCGR3A",
                       "CD14",
                       "CD4",
                       "LYVE1",
                       "VCAM1",
                       "COL6A2",
                       "DLK1",
                       "PDGFRB",
                       "MCAM",
                       "ACTA2",
                       "MYH11",
                       "DKK1",
                       "IGFBP1",
                       "VIM",
                       "CD59",
                       "TFAP2A",
                       "TFAP2C",
                       "GATA2",
                       "GATA3",
                       "TP63",
                       "TEAD4",
                       "ITGA6",
                       "BCAM",
                       "MKI67",
                       "ITGA2",
                       "TAGLN",
                       "NOTCH2",
                       "ITGA5",
                       "HLA-G",
                       "ITGA1",
                       "CGB",
                       "ERVFRD-1"),
         assay     = "RNA",
         cols      = c("lightgrey", "red"),
         col.min   = 0,
         col.max   = 1,
         dot.min   = 0,
         dot.scale = 15) +
theme   (axis.text.x = element_text (angle = 90, hjust = 1))

Idents (Expanded_Trophoblasts) <- "celltype"
```

```{r, Figure S4A, expanded single-cell RNA-seq dotplot of CD99, LEP, and AMOTL2 ligands, receptors, and/or binding partners}
Expanded_Trophoblasts$celltype.sex <- paste  (Idents (Expanded_Trophoblasts), Expanded_Trophoblasts$Sex, sep = "_")
Expanded_Trophoblasts$celltype     <- Idents (Expanded_Trophoblasts)

Idents (Expanded_Trophoblasts) <- "celltype.sex"

my_levels <- c("CTB_Male",               "CTB_Female",
               "cCTB_Male",              "cCTB_Female",
               "EVT_Male",               "EVT_Female",
               "SCTp_Male",              "SCTp_Female",
               "Macrophage_Male",        "Macrophage_Female",
               "uNK_Male",               "uNK_Female",
               "cNK_Male",               "cNK_Female",
               "Epiglandular cell_Male", "Epiglandular cell_Female",
               "Stromal cell_Male",      "Stromal cell_Female",
               "Endothelial cell_Male",  "Endothelial cell_Female")

Idents (Expanded_Trophoblasts) <- factor (Expanded_Trophoblasts$celltype.sex, levels = my_levels)

DotPlot (Expanded_Trophoblasts,
         features  = c("CD99",
                       "PILRA",
                       "PILRB",
                       "LEP",
                       "LEPR",
                       "AMOTL2",
                       "MAGI1",
                       "CDH5"),
         assay     = "RNA",
         cols      = c("lightgrey", "red"),
         col.min   = 0,
         col.max   = 1,
         dot.min   = 0,
         dot.scale = 15) +
theme   (axis.text.x = element_text (angle = 90, hjust = 1))

Idents (Expanded_Trophoblasts) <- "celltype"
```
