---
title: "CODE FOR FIGURE 1 AND SUPPLEMENTAL FIGURE 1"
author: "Matthew J Shannon"
date: "2024-12-17"
output: html_document
---

# Load in all dependencies

```{r, include = FALSE}
if (!requireNamespace ("BiocManager", quietly = TRUE))
    install.packages  ("BiocManager")
BiocManager::install  ("Rsamtools")
BiocManager::install  ("GenomicAlignments")
BiocManager::install  ("reshape2")
BiocManager::install  ("limma")
BiocManager::install  ("tidyverse")
BiocManager::install  ("reshape")
BiocManager::install  ("sva")
BiocManager::install  ("DESeq2")
BiocManager::install  ("biomaRt")
BiocManager::install  ("readxl")
BiocManager::install  ("apeglm")
BiocManager::install  ("clusterProfiler")
BiocManager::install  ("org.Hs.eg.db")
BiocManager::install  ("ggpubr")
BiocManager::install  ("ggpubr")
BiocManager::install  ("pvclust")
BiocManager::install  ('Seurat')
library               (Rsamtools)
library               (GenomicAlignments)
library               (GenomicRanges)
library               (tidyverse)
library               (knitr)
library               (readxl)
library               (edgeR)
library               (reshape2)
library               (limma)
library               (pheatmap)
library               (DESeq2)
library               (ggplot2)
library               (apeglm)
library               (grid)
library               (gridExtra)
library               (sva)
library               (png)
library               (biomaRt)
library               (RColorBrewer)
library               (ggpubr)
library               (pvclust)
library               (Seurat)
gc                    ()
```

# Load the necessary datasets

```{r}
# Full bulk RNA-seq object
load         (file = "~/Desktop/FULLdata.Rdata")

# Trophoblast single-cell object
load         (file = "~/Desktop/Trophoblast.Rdata")
DefaultAssay (Trophoblasts) <- "integrated"
```

# Figure 1

```{r, Figure 1D, single-cell RNA-seq trophoblast UMAP}
UMAPPlot (Trophoblasts,
          pt.size  = 2,
          split.by = "Sex",
          label    = FALSE,
          cols     = c('CTB'  = 'darkorange1',
                       'cCTB' = 'red3',
                       'EVT'  = 'forestgreen',
                       'SCTp' = 'dodgerblue4'))
```

```{r, Figure 1D, single-cell RNA-seq trophoblast cell proportions}
Trophoblasts$celltype.sex    <- paste       (Idents (Trophoblasts), Trophoblasts$Sex, sep = "_")
Trophoblasts$celltype        <- Idents      (Trophoblasts)

Idents (Trophoblasts) <- "celltype.sex"

ggplot             (Trophoblasts@meta.data, aes (x = c(Idents (Trophoblasts)), fill = Trophoblasts$celltype)) +
theme_classic2     () +
geom_bar           () +
theme              (text         = element_text (size   = 35)) +
theme              (axis.text    = element_text (size   = 35)) +
theme              (axis.title.x = element_text (face   = "bold")) +
theme              (axis.title.y = element_text (face   = "bold")) +
theme              (legend.text  = element_text (size   = 15)) +
theme              (axis.text.y  = element_text (angle  = 90)) +
theme              (axis.text.y  = element_text (hjust  = 0.5)) +
theme              (axis.text.x  = element_text (colour = "black")) +
theme              (axis.text.y  = element_text (colour = "black")) +
ylab               ("Cell Proportion")
```

```{r, Figure 1E, single-cell RNA-seq dotplot of trophoblast subtype-associated genes}
DotPlot (Trophoblasts,
         features  = c("TP63",
                       "TEAD4",
                       "ITGA6",
                       "BCAM",
                       "MKI67",
                       "ITGA2",
                       "TAGLN",
                       "SOX15",
                       "NOTCH2",
                       "ITGA5",
                       "HLA-G",
                       "ITGA1",
                       "CGB",
                       "ERVFRD-1"),
         assay     = "RNA",
         cols      = c("lightgrey", "red"),
         col.min   = 0,
         col.max   = 1,
         dot.min   = 0,
         dot.scale = 15) +
theme   (axis.text.x = element_text (angle = 90, hjust = 1))

Idents (Trophoblasts) <- "celltype"
```

```{r, Figure 1F, bulk (full) RNA-seq dataset PCA analysis}
sprDat <- t   (scale (t (fltlogcountData)))
          str (sprDat, max.level = 0, give.attr = FALSE)

sprDat <- round (data.frame (avgBefore = rowMeans (fltlogcountData),
                             avgAfter  = rowMeans (sprDat),
                             varBefore = apply    (fltlogcountData, 1, var),
                             varAfter  = apply    (sprDat, 1, var)), 2)

pcs <- prcomp (sprDat, scale = FALSE)

plot    (pcs, main = "Amount of Variance for Each Principle Component", xlab = "Principle Components")
summary (pcs)
```

```{r, Figure 1F, bulk (full) RNA-seq data - heat scree plot function creation}
heat_scree_plot <- function (Loadings, Importance, Num, Order) {
                             adjust       <- 1 - Importance [1]
                             pca_adjusted <-     Importance [2 : length (Importance)] / adjust
                             pca_df       <- data.frame (adjusted_variance = pca_adjusted, PC = seq (1 : length (pca_adjusted)))
  
                             scree <- ggplot             (pca_df [which (pca_df$PC < Num), ],
                                                          aes (PC, adjusted_variance)) +
                                      geom_bar           (stat = "identity", color = "black", fill = "grey") +
                                      theme_bw           () +
                                      theme              (axis.text   = element_text (size = 7),
                                                          axis.title  = element_text (size = 15),
                                                          plot.margin = unit (c (1, 1.5, 0.2, 2.25), "cm")) +
                                      ylab               ("Variance") +
                                      scale_x_continuous (breaks = seq (1, Num, 1))
  
#### Heat
  ## correlate meta with PCS
  ## Run anova of each PC on each meta data variable
  
aov_PC_meta          <- lapply (1 : ncol (meta_categorical),
                                function (covar) sapply (1 : ncol (Loadings),
                                function (PC) summary (aov (Loadings [ , PC] ~ meta_categorical [ , covar])) [[1]]$"Pr(>F)" [1]))
 names (aov_PC_meta) <- colnames      (meta_categorical)
        aov_PC_meta  <- do.call       (rbind, aov_PC_meta)
        aov_PC_meta  <- as.data.frame (aov_PC_meta)
  
# adjust
  aov_PC_meta_adjust <- aov_PC_meta [ , 2 : ncol (aov_PC_meta)]
  
# reshape
  avo                <- aov_PC_meta_adjust [ , 1 : (Num - 1)]
  avo_heat_num       <- apply         (avo, 2, as.numeric)
  avo_heat           <- as.data.frame (avo_heat_num)
  colnames(avo_heat) <- sapply        (1 : (Num - 1), function (x) paste ("PC", x, sep = ""))
  avo_heat$meta      <- rownames      (avo)
  avo_heat_melt      <- melt          (avo_heat, id = c("meta"))
  
# cluster meta data
  ord                  <- Order
  meta_var_order       <- unique (avo_heat_melt$meta) [rev (ord)]
  avo_heat_melt$meta   <- factor (avo_heat_melt$meta, levels = meta_var_order)
  avo_heat_melt$Pvalue <- sapply (1 : nrow (avo_heat_melt),
                                  function (x) if (avo_heat_melt$value [x] <= 0.001) {"<=0.001"} else {
                                               if (avo_heat_melt$value [x] <= 0.01)  {"<=0.01"}  else {
                                               if (avo_heat_melt$value [x] <= 0.05)  {"<=0.05"}  else {">0.05"}}})
  
  heat <- ggplot            (avo_heat_melt, aes (variable, meta, fill = Pvalue)) +
          geom_tile         (color = "black", size = 0.5) +
          theme_gray        (8) +
          scale_fill_manual (values               = c("#084594", "#4292c6", "#9ecae1", "#deebf7")) +
          theme             (axis.text            = element_text (size = 7, color = "black"),
                             axis.text.x          = element_text (),
                             axis.title           = element_text (size = 15),
                             legend.text          = element_text (size = 14),
                             legend.title         = element_text (size = 12),
                             legend.position      = "top",
                             legend.justification = c(1, 0),
                             plot.margin          = unit (c(0, 2.25, 1, 1), "cm")) +
          xlab              ("Principal Component") +
          ylab              (NULL)
  
  grid.arrange (scree, heat, ncol = 1, widths = c(4), heights = c(2, 4)) }
```

```{r, Figure 1F, bulk (full) RNA-seq data - PCA heatmap}
PCA_full <- princomp (fltlogcountData, center = TRUE, scale = TRUE, cor = FALSE)

#scaling is not necessary for a normalized dataset

Loadings     <- as.data.frame (unclass (PCA_full$loadings))
vars         <- PCA_full$sdev^2
Importance   <- vars / sum (vars)
adjust       <- 1 - Importance [1]
pca_adjusted <-     Importance [2 : length (Importance)] / adjust
pca_df       <- data.frame (adjusted_variance = pca_adjusted, PC = seq (1 : length (pca_adjusted)))

          metaData
colnames (metaData)
str      (metaData)


meta_categorical  <- metaData[ , c("GA", "Sex", "MA","Smoking Status", "BMI", "Serum CRP", "AH Drug Use", "AI Drug Use", "Batch")]


# Specify the number of PCs you want shown
Num <- 35 # should be about equal to the number of samples in your dataset; for large datasets, you can opt to just see the top PCs 

# Designate what order you want the variables to appear
Order <- c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)

#Apply function on PCA results, pulls in the meta data and beta values from above
heat_scree_plot (Loadings, Importance, Num, Order)

rm (pcs)
rm (sprDat)
```

```{r, Figure 1G, bulk (full) RNA-seq data MDS plots}
col.cell <- c("#d25e8cff","#5f8cd3ff", "#784421ff")[metaData$Sex]

levels     (metaData$Sex)
data.frame (metaData$Sex, col.cell)

plotMDS    (FULLdata, pch = 19, col = col.cell, xlim = c(-3, 3), ylim = c(-3, 3), dim.plot = c(1, 2))
plotMDS    (FULLdata, pch = 19, col = col.cell, xlim = c(-3, 3), ylim = c(-3, 3), dim.plot = c(3, 4))
plotMDS    (FULLdata, pch = 19, col = col.cell, xlim = c(-3, 3), ylim = c(-3, 3), dim.plot = c(5, 6))


col.cell <- c("#71c837ff", "#ffcc00ff")[metaData$GA]

levels     (metaData$GA)
data.frame (metaData$GA, col.cell)

plotMDS    (FULLdata, pch = 19, col = col.cell, xlim = c(-3, 3), ylim = c(-3, 3), dim.plot = c(1, 2))
plotMDS    (FULLdata, pch = 19, col = col.cell, xlim = c(-3, 3), ylim = c(-3, 3), dim.plot = c(3, 4))
plotMDS    (FULLdata, pch = 19, col = col.cell, xlim = c(-3, 3), ylim = c(-3, 3), dim.plot = c(5, 6))
```

# Supplemental figure 1

```{r, Figure S1B, bulk (full) RNA-seq data}
corexpressionMatrix <- cor (fltlogcountData, method = "spear", use = "pair")
metaData            <- metaData %>% column_to_rownames ("Sample")

pheatmap (corexpressionMatrix,
          cluster_rows             = TRUE,
          show_colnames            = TRUE,
          show_rownames            = TRUE,
          scale                    = "none",
          clustering_method        = "ward.D2",
          clustering_distance_cols = "euclidean",
          main                     = "Clustering Heatmap",
          annotation               = metaData[ , c("Sex", "GA", "BMI", "CRP", "Smoke", "Batch")])

rm (corexpressionMatrix)
```

```{r, Figure S1C, bulk (full) RNA-seq data MDS plots}
col.cell <- c("#37c89bff", "#ffa100ff")[metaData$MA]
levels     (metaData$MA)
data.frame (metaData$MA, col.cell)
plotMDS    (FULLdata,   pch = 19, col = col.cell, xlim = c(-3, 3), ylim = c(-3, 3), dim.plot = c(1, 2))

col.cell <- c("#f2f2f2ff", "#6c5d53ff")[metaData$Smoke]
levels     (metaData$Smoke)
data.frame (metaData$Smoke, col.cell)
plotMDS    (FULLdata,   pch = 19, col = col.cell, xlim = c(-3, 3), ylim = c(-3, 3), dim.plot = c(1, 2))

col.cell <- c("#51bfffff","#fff84cff", "#ff60b3ff")[metaData$BMI]
levels     (metaData$BMI)
data.frame (metaData$BMI, col.cell)
plotMDS    (FULLdata,   pch = 19, col = col.cell, xlim = c(-3, 3), ylim = c(-3, 3), dim.plot = c(1, 2))

col.cell <- c("#ff0000ff","#ffff00ff", "#00ffffff", "#0000ffff", "#00ff00ff")[metaData$Batch]
levels     (metaData$Batch)
data.frame (metaData$Batch, col.cell)
plotMDS    (FULLdata,   pch = 19, col = col.cell, xlim = c(-3, 3), ylim = c(-3, 3), dim.plot = c(1, 2))

rm (col.cell)
```

```{r, Figure S1D, pseudo-bulk single-cell RNA-seq data PCA plots}
cCTB     <- subset               (Trophoblasts, idents = c("cCTB"))
cCTB.avg <- AverageExpression    (cCTB,     return.seurat = TRUE)
cCTB.avg <- FindVariableFeatures (cCTB.avg, selection.method = "vst", nfeatures = 1000)
cCTB.avg <- RunPCA               (cCTB.avg, npcs = 8)

PCAPlot (cCTB.avg, pt.size = 4, label = TRUE) + NoLegend ()

EVT     <- subset               (Trophoblasts, idents = c("EVT"))
EVT.avg <- AverageExpression    (EVT,     return.seurat = TRUE)
EVT.avg <- FindVariableFeatures (EVT.avg, selection.method = "vst", nfeatures = 1000)
EVT.avg <- RunPCA               (EVT.avg, npcs = 8)

PCAPlot (EVT.avg, pt.size = 4, label = TRUE) + NoLegend ()
```

```{r, Figure S1E, male and female single-cell RNA-seq PCC heatmap}
Trophoblasts$celltype.sex    <- paste       (Idents (Trophoblasts), Trophoblasts$Sex, sep = "_")
Trophoblasts$celltype        <- Idents      (Trophoblasts)

       Idents (Trophoblasts) <- "celltype.sex"
table (Idents (Trophoblasts))

av.exp  <- AverageExpression  (Trophoblasts)$RNA
av.exp  <- as.data.frame      (av.exp)
cor.exp <- cor                (av.exp)
cor.exp <- as.data.frame      (cor.exp)
cor.exp <- rownames_to_column (cor.exp, "x")
cor.df  <- tidyr::gather      (cor.exp, y, correlation, c('CTB-Male',  'CTB-Female',
                                                          'cCTB-Male', 'cCTB-Female',
                                                          'EVT-Male',  'EVT-Female',
                                                          'SCTp-Male', 'SCTp-Female'))
cor.df # PCC values

my_levels <- c("CTB-Female", "CTB-Male", "cCTB-Female", "cCTB-Male", "EVT-Female", "EVT-Male", "SCTp-Female", "SCTp-Male")
cor.df$x  <- factor (cor.df$x, levels = my_levels)
cor.df$y  <- factor (cor.df$y, levels = my_levels)

ggplot               (cor.df, aes (x, y, fill = correlation)) +
geom_tile            () +
scale_fill_gradientn (limits = c(0, 1), colours = colorRampPalette (rev (brewer.pal (n = 8, name = "RdYlBu"))) (100))
```

```{r, Figure S1F, single-cell RNA-seq data feature plots}
DefaultAssay (Trophoblasts) <- "RNA"

FeaturePlot  (Trophoblasts, features = c("NOTCH2"), split.by = "Sex", label = FALSE, pt.size = 2, min.cutoff = 0, keep.scale = "all", order = TRUE, cols = c("grey", "red"))
FeaturePlot  (Trophoblasts, features = c("ITGA5"),  split.by = "Sex", label = FALSE, pt.size = 2, min.cutoff = 0, keep.scale = "all", order = TRUE, cols = c("grey", "red"))
FeaturePlot  (Trophoblasts, features = c("HLA-G"),  split.by = "Sex", label = FALSE, pt.size = 2, min.cutoff = 0, keep.scale = "all", order = TRUE, cols = c("grey", "red"))
FeaturePlot  (Trophoblasts, features = c("ITGA1"),  split.by = "Sex", label = FALSE, pt.size = 2, min.cutoff = 0, keep.scale = "all", order = TRUE, cols = c("grey", "red"))

DefaultAssay (Trophoblasts) <- "integrated"
```

```{r, Figure S1G, single-cell RNA-seq data}
cCTB     <- subset            (cCTB, idents = c("B_P277", "B_P89", "B_P21", "B_P130", "B_P244", "B_P295", "VT_P10", "VT_P11", "VT_P12"))
cCTB.avg <- AverageExpression (cCTB, group.by = 'ID')$RNA

cCTB.avg ["NOTCH2", ]   # exported and plotted using GraphPad Prism
cCTB.avg ["ITGA5", ]    # exported and plotted using GraphPad Prism
cCTB.avg ["HLA-G", ]    # exported and plotted using GraphPad Prism
cCTB.avg ["ITGA1", ]    # exported and plotted using GraphPad Prism

EVT     <- subset            (EVT, idents = c("B_P277", "B_P89", "B_P21", "B_P130", "B_P244", "B_P295", "B_P899", "VT_P9", "VT_P10", "VT_P11", "VT_P12"))
EVT.avg <- AverageExpression (EVT,  group.by = 'ID')$RNA

EVT.avg ["NOTCH2", ]   # exported and plotted using GraphPad Prism
EVT.avg ["ITGA5", ]    # exported and plotted using GraphPad Prism
EVT.avg ["HLA-G", ]    # exported and plotted using GraphPad Prism
EVT.avg ["ITGA1", ]    # exported and plotted using GraphPad Prism
```

```{r, Figure S1G, bulk (minus sample E12) RNA-seq  data prep}
FULLdata$P185  <- as.integer (FULLdata$P185)
FULLdata$P156  <- as.integer (FULLdata$P156)
FULLdata$P223  <- as.integer (FULLdata$P223)
FULLdata$P766  <- as.integer (FULLdata$P766)
FULLdata$P280  <- as.integer (FULLdata$P280)
FULLdata$P5    <- as.integer (FULLdata$P5)
FULLdata$P15   <- as.integer (FULLdata$P15)
FULLdata$P250  <- as.integer (FULLdata$P250)
FULLdata$P568  <- as.integer (FULLdata$P568)
FULLdata$P193  <- as.integer (FULLdata$P193)
FULLdata$P217  <- as.integer (FULLdata$P217)
FULLdata$P181  <- as.integer (FULLdata$P181)
FULLdata$P247  <- as.integer (FULLdata$P247)
FULLdata$P252  <- as.integer (FULLdata$P252)
FULLdata$P456  <- as.integer (FULLdata$P456)
FULLdata$P979  <- as.integer (FULLdata$P979)
FULLdata$P984  <- as.integer (FULLdata$P984)
FULLdata$P996  <- as.integer (FULLdata$P996)
FULLdata$P1003 <- as.integer (FULLdata$P1003)
FULLdata$P1004 <- as.integer (FULLdata$P1004)
FULLdata$P1024 <- as.integer (FULLdata$P1024)
FULLdata$P1047 <- as.integer (FULLdata$P1047)
FULLdata$P1051 <- as.integer (FULLdata$P1051)
FULLdata$P122  <- as.integer (FULLdata$P122)
FULLdata$P132  <- as.integer (FULLdata$P132)
FULLdata$P219  <- as.integer (FULLdata$P219)
FULLdata$P126  <- as.integer (FULLdata$P126)
FULLdata$P300  <- as.integer (FULLdata$P300)
FULLdata$P1012 <- as.integer (FULLdata$P1012)
FULLdata$P1015 <- as.integer (FULLdata$P1015)
FULLdata$P1019 <- as.integer (FULLdata$P1019)
FULLdata$P1025 <- as.integer (FULLdata$P1025)
FULLdata$P1029 <- as.integer (FULLdata$P1029)
FULLdata$P1037 <- as.integer (FULLdata$P1037)
FULLdata$P1041 <- as.integer (FULLdata$P1041)
counts          <- as.matrix  (FULLdata)
meta            <- as.matrix  (metaData)
```

```{r, Figure S1G, bulk (minus sample E12) RNA-seq data}
dds <- DESeqDataSetFromMatrix (countData = counts,
                               colData   = meta,
                               design    = ~ Batch + GA2 + Sex)

keep <- rowSums (counts (dds) >= 1) > 3
dds  <- dds [keep, ]
dds  <- DESeq   (dds)

      NOTCH2 <- plotCounts (dds, gene = "NOTCH2", intgroup = c("Sex", "GA"), returnData = TRUE)
view (NOTCH2)   # exported and plotted using GraphPad Prism

      ITGA5  <- plotCounts (dds, gene = "ITGA5",  intgroup = c("Sex", "GA"), returnData = TRUE)
view (ITGA5)    # exported and plotted using GraphPad Prism

      HLA-G  <- plotCounts (dds, gene = "HLA-G",  intgroup = c("Sex", "GA"), returnData = TRUE)
view (HLA-G)    # exported and plotted using GraphPad Prism

      ITGA1  <- plotCounts (dds, gene = "ITGA1",  intgroup = c("Sex", "GA"), returnData = TRUE)
view (ITGA1)    # exported and plotted using GraphPad Prism

rm (dds)
rm (keep)
```
