---
title: "CODE FOR FIGURE 2 AND SUPPLEMENTAL FIGURE 2"
author: "Matthew J Shannon"
date: "2024-12-17"
output: html_document
---

# Load in all dependencies

```{r, include = FALSE}
if (!requireNamespace ("BiocManager", quietly = TRUE))
    install.packages  ("BiocManager")
BiocManager::install  ("Rsamtools")
BiocManager::install  ("GenomicAlignments")
BiocManager::install  ("reshape2")
BiocManager::install  ("limma")
BiocManager::install  ("tidyverse")
BiocManager::install  ("reshape")
BiocManager::install  ("sva")
BiocManager::install  ("DESeq2")
BiocManager::install  ("biomaRt")
BiocManager::install  ("readxl")
BiocManager::install  ("apeglm")
BiocManager::install  ("clusterProfiler")
BiocManager::install  ("org.Hs.eg.db")
BiocManager::install  ("ggpubr")
BiocManager::install  ("ggpubr")
BiocManager::install  ("pvclust")
BiocManager::install  ('Seurat')
library               (Rsamtools)
library               (GenomicAlignments)
library               (GenomicRanges)
library               (tidyverse)
library               (knitr)
library               (readxl)
library               (edgeR)
library               (reshape2)
library               (limma)
library               (pheatmap)
library               (DESeq2)
library               (ggplot2)
library               (apeglm)
library               (grid)
library               (gridExtra)
library               (sva)
library               (png)
library               (biomaRt)
library               (RColorBrewer)
library               (ggpubr)
library               (pvclust)
library               (Seurat)
gc                    ()
```

# Load the necessary datasets

```{r}
# Full bulk RNA-seq object
load         (file = "~/Desktop/FULLdata.Rdata")

# Small bulk RNA-seq (minus samples with a GA = 9.0) object
load         (file = "~/Desktop/SMALLdata.Rdata")

# Trophoblast single-cell object
load         (file = "~/Desktop/Trophoblast.Rdata")
DefaultAssay (Trophoblasts) <- "integrated"
```

# Supplemental figure 2

```{r}
meta <- as.matrix (metaData)
```

```{r, Figure 2A and Figure S2, bulk (full) RNA-seq data}
dds         <- DESeqDataSetFromMatrix            (countData = round (FULLdata), colData = meta, design = ~1)
vsd         <- varianceStabilizingTransformation (dds, blind = FALSE)
topVarGenes <- head                              (order (rowVars (assay (vsd)), decreasing = TRUE), 50)
mat         <- FULLdata[topVarGenes, ]
mat         <- mat - rowMeans (mat)
anno        <- as.data.frame  (colData (vsd)[ , c("GA","Sex", "MA", "BMI", "Smoke", "AI",  "AH")])

pheatmap (mat, annotation_col = anno, cluster_cols = TRUE, cluster_rows = TRUE) # Dendrogram used for Fig 2A and the heatmap is used for Fig S2
```

# Figure 2 - indicate dataset used in {r; example = r, Figure S1A, bulk RNA-seq dataset}

```{r, Figure 2B, bulk (full dataset minus samples with a GA within week 9 of gestation) GA-based differential expression analysis prep}
SMALLdata$P122  <- as.integer (SMALLdata$P122)
SMALLdata$P132  <- as.integer (SMALLdata$P132)
SMALLdata$P185  <- as.integer (SMALLdata$P185)
SMALLdata$P223  <- as.integer (SMALLdata$P223)
SMALLdata$E12   <- as.integer (SMALLdata$E12)
SMALLdata$P219  <- as.integer (SMALLdata$P219)
SMALLdata$P280  <- as.integer (SMALLdata$P280)
SMALLdata$P5    <- as.integer (SMALLdata$P5)
SMALLdata$P250  <- as.integer (SMALLdata$P250)
SMALLdata$P568  <- as.integer (SMALLdata$P568)
SMALLdata$P193  <- as.integer (SMALLdata$P193)
SMALLdata$P126  <- as.integer (SMALLdata$P126)
SMALLdata$P300  <- as.integer (SMALLdata$P300)
SMALLdata$P181  <- as.integer (SMALLdata$P181)
SMALLdata$P247  <- as.integer (SMALLdata$P247)
SMALLdata$P252  <- as.integer (SMALLdata$P252)
SMALLdata$P979  <- as.integer (SMALLdata$P979)
SMALLdata$P984  <- as.integer (SMALLdata$P984)
SMALLdata$P996  <- as.integer (SMALLdata$P996)
SMALLdata$P1004 <- as.integer (SMALLdata$P1004)
SMALLdata$P1012 <- as.integer (SMALLdata$P1012)
SMALLdata$P1015 <- as.integer (SMALLdata$P1015)
SMALLdata$P1019 <- as.integer (SMALLdata$P1019)
SMALLdata$P1024 <- as.integer (SMALLdata$P1024)
SMALLdata$P1025 <- as.integer (SMALLdata$P1025)
SMALLdata$P1029 <- as.integer (SMALLdata$P1029)
SMALLdata$P1037 <- as.integer (SMALLdata$P1037)
SMALLdata$P1041 <- as.integer (SMALLdata$P1041)
SMALLdata$P1047 <- as.integer (SMALLdata$P1047)
SMALLdata$P1051 <- as.integer (SMALLdata$P1051)
counts          <- as.matrix  (SMALLdata)
meta            <- as.matrix  (metaData)
```

```{r, Figure 2B, bulk (full dataset minus samples with a GA within week 9 of gestation) GA-based differential expression analysis}
dds     <- DESeqDataSetFromMatrix (countData = counts, colData = meta, design = ~ Batch + Sex + GA2)
keep    <- rowSums                (counts (dds) >= 1) > 3
dds     <- dds [keep, ]

dds$GA2 <- factor  (dds$GA2, levels = c("0", "1"))
dds     <- DESeq   (dds)
res     <- results (dds)
resSig  <- subset  (resOrdered, padj < 0.05)

view (resSig)
rm   (keep)
```

```{r, Figure 2B, bulk (full dataset minus samples with a GA within week 9 of gestation) volcano plot}
EnhancedVolcano (res,
                 lab             = rownames (res),
                 x               = "log2FoldChange",
                 y               = "pvalue",
                 FCcutoff        = 1.00,
                 pCutoff         = 10e-5,
                 pointSize       = 2.0,
                 col             = c('grey', 'grey', 'grey', 'red'),
                 colAlpha        = 1,
                 xlim            = c(-11, 11),
                 gridlines.major = FALSE,
                 drawConnectors  = FALSE, 
                 gridlines.minor = FALSE) + ggtitle (NULL)
```

```{r, Figure 2C, single-cell RNA-seq dotplot of the top 20 p-value ranked results from Fig 2B}
Trophoblasts$celltype.sex    <- paste       (Idents (Trophoblasts), Trophoblasts$Sex, sep = "_")
Trophoblasts$celltype        <- Idents      (Trophoblasts)

Idents (Trophoblasts) <- "celltype.sex"

DotPlot (Trophoblasts,
         features  = c("PAPPA",
                       "LAYN",
                       "CSH2",
                       "PSG2",
                       "BGN",
                       "PDGFB",
                       "EFHD1",
                       "MFSD2A",
                       "MMP11",
                       "SCARB1",
                       "EBI3",
                       "FBLN1",
                       "CCDC69",
                       "PAPPA2",
                       "ST3GAL5",
                       "PSG5",
                       "CSH1",
                       "HES2",
                       "HEXB",
                       "CABLES1"),
         assay     = "RNA",
         cols      = c("lightgrey", "red"),
         col.min   = 0,
         col.max   = 1,
         dot.min   = 0,
         dot.scale = 15) +
theme   (axis.text.x = element_text (angle = 90, hjust = 1))

Idents  (Trophoblasts) <- "celltype"
```

```{r, Figure 2D, bulk (full dataset minus samples with a GA within week 9 of gestation) gene ontology analysis and plots}
# Create background dataset for hypergeometric testing using all genes tested for significance in the results
all_genes <- as.character (rownames (res))

# Extract significant results
signif_YOUNG <- res[res$padj < 0.05 & res$log2FoldChange < 0.00 & !is.na (res$padj), ]
signif_YOUNG <- as.character (rownames (signif_YOUNG))

# Run GO enrichment analysis
YOUNG <- enrichGO (gene          = signif_YOUNG,
                   universe      = all_genes,
                   keyType       = "SYMBOL",
                   OrgDb         = "org.Hs.eg.db",
                   ont           = "BP",
                   pAdjustMethod = "BH",
                   qvalueCutoff  = 0.05)

# Output results from GO analysis to a table
cluster_summary_YOUNG <- data.frame (YOUNG)

# Extract significant results
signif_OLD <- res[res$padj < 0.05 & res$log2FoldChange > 0.00 & !is.na (res$padj), ]
signif_OLD <- as.character (rownames (signif_OLD))

# Run GO enrichment analysis
OLD <- enrichGO (gene          = signif_OLD,
                 universe      = all_genes,
                 keyType       = "SYMBOL",
                 OrgDb         = "org.Hs.eg.db",
                 ont           = "BP",
                 pAdjustMethod = "BH",
                 qvalueCutoff  = 0.05)

# Output results from GO analysis to a table
cluster_summary_OLD <- data.frame (OLD)

# plot
      options (enrichplot.colours = c("#71c837ff", "grey"))
P1 <- dotplot (YOUNG, showCategory = 20) + xlim (0, 0.15)

      options (enrichplot.colours = c("#ffcc00ff", "grey"))
P2 <- dotplot (OLD,   showCategory = 20) + xlim (0, 0.15)

P1 + P2
```
